astronomer_starship/compat/starship_compatability.py

```diff
@@ -992,29 +994,31 @@ def delete_xcom(self, **kwargs):
         res.status_code = 409
         raise NotImplementedError()
 
-    def insert_directly(self, table_name, items):
+    def insert_directly(self, table_name, items, should_remove_id_field: bool = True):
         from sqlalchemy.exc import InvalidRequestError
         from sqlalchemy import MetaData
+        from airflow import __version__
         import pickle
+        import re
+
+        [major, _] = re.sub(r"[^0-9.]", "", __version__).split(".", maxsplit=1)
```

Comment:
instead of manually parsing versions with regex, I recommend to use [packaging](https://packaging.pypa.io/en/stable/version.html) instead.

```
>>> from packaging.version import Version
>>> v = Version("2.10.5+astro.3")
>>> v.major
2
>>>
```